mm2src/coins/utxo/utxo_tests.rs

```diff
@@ -3867,3 +3753,55 @@ fn test_electrum_balance_deserializing() {
     assert_eq!(actual.confirmed, i128::MIN);
     assert_eq!(actual.unconfirmed, i128::MAX);
 }
+
+#[test]
+fn test_electrum_display_balances() {
+    let rpc_client = electrum_client_for_test(RICK_ELECTRUM_ADDRS);
+    block_on(utxo_common_tests::test_electrum_display_balances(&rpc_client));
+}
+
+#[test]
+fn test_native_display_balances() {
+    let unspents = vec![
+        NativeUnspent {
+            address: "RG278CfeNPFtNztFZQir8cgdWexVhViYVy".to_owned(),
+            amount: "4.77699".into(),
+            ..NativeUnspent::default()
+        },
+        NativeUnspent {
+            address: "RJeDDtDRtKUoL8BCKdH7TNCHqUKr7kQRsi".to_owned(),
+            amount: "0.77699".into(),
+            ..NativeUnspent::default()
+        },
+        NativeUnspent {
+            address: "RQHn9VPHBqNjYwyKfJbZCiaxVrWPKGQjeF".to_owned(),
+            amount: "0.99998".into(),
+            ..NativeUnspent::default()
+        },
+        NativeUnspent {
+            address: "RG278CfeNPFtNztFZQir8cgdWexVhViYVy".to_owned(),
+            amount: "1".into(),
+            ..NativeUnspent::default()
+        },
+    ];
+
+    NativeClient::list_unspent_impl
+        .mock_safe(move |_, _, _, _| MockResult::Return(Box::new(futures01::future::ok(unspents.clone()))));
+
+    let rpc_client = native_client_for_test();
+
+    let expected: Vec<(Address, BigDecimal)> = vec![
```

Comment:
The `expected` value is used in the line below
https://github.com/KomodoPlatform/atomicDEX-API/blob/7224b85e39a2406b5b9e8cb9a8e69cd59fda514f/mm2src/coins/utxo/utxo_tests.rs#L3793-L3800

This could be done in more lines:
```rust
let addresses = vec![
        "RG278CfeNPFtNztFZQir8cgdWexVhViYVy".into(),
        "RYPz6Lr4muj4gcFzpMdv3ks1NCGn3mkDPN".into(),
        "RJeDDtDRtKUoL8BCKdH7TNCHqUKr7kQRsi".into(),
        "RQHn9VPHBqNjYwyKfJbZCiaxVrWPKGQjeF".into(),
    ];

    let addresses = expected.iter().map(|(address, _)| address.clone()).collect();
    let actual = rpc_client
        .display_balances(addresses, TEST_COIN_DECIMALS)
        .wait()
        .unwrap();

    let expected: Vec<(Address, BigDecimal)> = vec![
        ("RG278CfeNPFtNztFZQir8cgdWexVhViYVy".into(), BigDecimal::from(5.77699)),
        ("RYPz6Lr4muj4gcFzpMdv3ks1NCGn3mkDPN".into(), BigDecimal::from(0)),
        ("RJeDDtDRtKUoL8BCKdH7TNCHqUKr7kQRsi".into(), BigDecimal::from(0.77699)),
        ("RQHn9VPHBqNjYwyKfJbZCiaxVrWPKGQjeF".into(), BigDecimal::from(0.99998)),
    ];
    assert_eq!(actual, expected);
```
What do you think?

Reply:
Looks okay. You can consider also the following:
```diff
fn test_native_display_balances() {
     let rpc_client = native_client_for_test();

-    let expected: Vec<(Address, BigDecimal)> = vec![
-        ("RG278CfeNPFtNztFZQir8cgdWexVhViYVy".into(), BigDecimal::from(5.77699)),
-        ("RYPz6Lr4muj4gcFzpMdv3ks1NCGn3mkDPN".into(), BigDecimal::from(0)),
-        ("RJeDDtDRtKUoL8BCKdH7TNCHqUKr7kQRsi".into(), BigDecimal::from(0.77699)),
-        ("RQHn9VPHBqNjYwyKfJbZCiaxVrWPKGQjeF".into(), BigDecimal::from(0.99998)),
+    let addresses: Vec<Address> = vec![
+        "RG278CfeNPFtNztFZQir8cgdWexVhViYVy".into(),
+        "RYPz6Lr4muj4gcFzpMdv3ks1NCGn3mkDPN".into(),
+        "RJeDDtDRtKUoL8BCKdH7TNCHqUKr7kQRsi".into(),
+        "RQHn9VPHBqNjYwyKfJbZCiaxVrWPKGQjeF".into(),
     ];

-    let addresses = expected.iter().map(|(address, _)| address.clone()).collect();
     let actual = rpc_client
-        .display_balances(addresses, TEST_COIN_DECIMALS)
+        .display_balances(addresses.clone(), TEST_COIN_DECIMALS)
         .wait()
         .unwrap();

+    let expected_balances: Vec<BigDecimal> = vec![
+        BigDecimal::from(5.77699 as f64),
+        BigDecimal::from(0 as i32),
+        BigDecimal::from(0.77699 as f64),
+        BigDecimal::from(0.99998 as f64),
+    ];
+
+    assert_eq!(addresses.len(), expected_balances.len());
+
+    let mut expected: Vec<(Address, BigDecimal)> = Vec::new();
+    addresses.iter().enumerate().for_each(|(index, address)| {
+        expected.push((*address, expected_balances[index]));
+    });
+
     assert_eq!(actual, expected);
 }
```

Reply:
Isn't my solution more compact and easy to read? 😅

Reply:
I just wanted to share a code flow from another aspect, all looks confirmable. The decision is up to you, I am fine with any of them :)

Reply:
Done